frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1935,17 +1978,41 @@ function ReaderHighlight:extendSelection()
         table.sort(positions, comparePositions)
         new_pos0 = positions[1]
         new_pos1 = positions[4]
-        local text_boxes = self.ui.document:getTextFromPositions(new_pos0, new_pos1)
-        new_text = text_boxes.text
-        new_pboxes = text_boxes.pboxes
+        local temp_pos0, temp_pos1
+        if new_pos0.page == new_pos1.page then -- single-page highlight
+            local text_boxes = self.ui.document:getTextFromPositions(new_pos0, new_pos1)
+            new_text = text_boxes.text
+            new_pboxes = text_boxes.pboxes
+            temp_pos0, temp_pos1 = new_pos0, new_pos1
+        else -- multi-page highlight
+            new_text = ""
+            new_pboxes = {}
+            ext = {} -- additional information saved in multi-page highlight
```

Comment:
May be add "one item per additional page" in the comment - if I understood that right.
Or - if I understand that more right: "one item per page, including the first one"